src/core/sensor/qgssensormanager.h

```diff
@@ -0,0 +1,127 @@
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSENSORMANAGER_H
+#define QGSSENSORMANAGER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsabstractsensor.h"
+
+#include <QObject>
+#include <QDomElement>
+
+class QgsProject;
+
+/**
+ * \ingroup core
+ * \class QgsSensorManager
+ *
+ * \brief Manages sensors.
+ *
+ * QgsSensorManager handles the storage, serializing and deserializing
+ * of sensors. Usually this class is not constructed directly, but
+ * rather accessed through a QgsProject via QgsProject::sensorManager().
+ *
+ * \since QGIS 3.32
+ */
+class CORE_EXPORT QgsSensorManager : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Returns a newly created QgsSensorManager linked to the specified \a project.
+     *
+     * The returned object is parented to the \a project.
+     */
+    static QgsSensorManager *createProjectBasedManager( QgsProject *project );
+
+    /**
+     * Constructor for QgsSensorManager, with the specified \a parent object.
+     */
+    explicit QgsSensorManager( QObject *parent SIP_TRANSFERTHIS = nullptr );
+
+    ~QgsSensorManager() override;
+
+    /**
+     * Deregisters and removes all sensors from the manager.
+     */
+    void clear();
+
+    /**
+     * Returns a list of pointers to all registered sensors.
+     */
+    QList<QgsAbstractSensor *> sensors() const;
+
+    /**
+     * Returns a registered sensor pointer matching a given \a id. If not
+     * matching sensor is found, a NULLPTR will be returned.
+     */
+    QgsAbstractSensor *sensor( const QString &id ) const;
+
+    /**
+     * Adds a \a sensor. Takes ownership of the sensor.
+     */
+    void addSensor( QgsAbstractSensor *sensor SIP_TRANSFER );
+
+    /**
+     * Removes a registered sensor matching a given \a id.
+     * \returns TRUE if a sensor was removed.
+     */
+    bool removeSensor( const QString &id );
+
+    /**
+     * Returns a list of registered sensor names.
+     */
+    QStringList sensorNames() const;
+
+    /**
+     * Returns the last captured data from a registered sensor matching a given \a name.
+     */
+    QgsAbstractSensor::SensorData sensorData( const QString &name ) const;
+
+    /**
+     * Returns the last captured data of all registered sensors.
+     */
+    QMap<QString, QgsAbstractSensor::SensorData> sensorsData() const;
+
+    /**
+     * Reads the manager's state from a DOM element, restoring all sensors
+     * present in the XML document.
+     * \see writeXml()
+     */
+    bool readXml( const QDomElement &element, const QDomDocument &document );
+
+    /**
+     * Returns a DOM element representing the state of the manager.
+     * \see readXml()
+     */
+    QDomElement writeXml( QDomDocument &document ) const;
+
+  private slots:
```

Comment:
I think we're missing some useful signals here -- eg one when new data is available (from any sensor) and one when new data is available for a specific sensor